acados/utils/external_function_generic.c

```diff
@@ -192,24 +190,72 @@ static int casadi_nnz(const int *sparsity)
 }
 
 
+static int casadi_is_dense(const int *sparsity)
+{
+    int idx, jj;
+    if (sparsity != NULL)
+    {
+        const int nrow = sparsity[0];
+        const int ncol = sparsity[1];
+        const int dense = sparsity[2];
+        if ((nrow<=0 )| (ncol<=0))
+            return 1;
+        if (dense)
+        {
+            return 1;
+        }
+        else
+        {
+            // check for casadi fake sparse
+            const int *idxcol = sparsity + 2;
+            const int *row = sparsity + ncol + 3;
+            for (jj = 0; jj < ncol; jj++)
+            {
+                if (idxcol[jj + 1] - idxcol[jj] != nrow)
```

Comment:
In principle, generally speaking in the sparse matrix format the elements do not need to be ordered within a column, so here just checking that each column has exactly `nrow` elements would fail if that is not the case.

In practice, is there any CasADi documentation about assuming column elements ordering in their sparse matrix format?

Reply:
This if only says that a matrix is not dense, if there is a structural zero element in a column.
The next `for`-`if` loop additionally checks that the indices are also increasing.
I do think that this next check should not be necessary in practice, but I did not find any guarantee on CasADi side and it is a one time offline check, so better to put it there.

Reply:
OK perfect then, maybe a comment explaining what the code does would be handy :)